nodejs/week3/homework/warmup/index.js

```diff
@@ -0,0 +1,26 @@
+const express = require("express");
+const app = express();
+
+app.get("/", (req, res) => res.send("nodejs week3 homework"));
+
+app.get('/calculator/multiply',  (req, res) => {
+    const values = Object.values(req.query).flat().reduce((acc, val) => (acc) * (val))
+    res.send(` ${values}`)
+});
+app.get('/calculator/add',  (req, res) => {
+    const values = Object.values(req.query).flat().reduce((acc, val) => ((acc)*1 + (val)*1))
+    res.send(` ${values}`)
+});
+app.get('/calculator/substract',  (req, res) => {
+    const values = Object.values(req.query).flat().reduce((acc, val) => 1*(acc - (val)))
+    res.send(` ${values}`)
+});
+app.get('/calculator/divide',  (req, res) => {
+    const values = Object.values(req.query).flat().reduce((acc, val) => 1*((acc) / (val)))
+    res.send(` ${values}`)
+});
+app.post('/calculator/add', (req, res) =>{
+    const values = Object.values(req.query).flat().reduce((acc, val) =>((acc)*1 + (val)*1))
```

Comment:
```suggestion
    const values = Object.values(req.body).flat().reduce((acc, val) =>((acc)*1 + (val)*1))
```
Have you tested it? 🙂

Reply:
Yes I did that I got that error (TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at C:\Users\shanh\Desktop\hyf-homework\nodejs\week3\homework\warmup\index.js:31:25)

Reply:
Will it work if you add app.use(express.urlencoded({extended: true}));?

Reply:
Thanks, I implemented the change